### PR TITLE
AWS Shield Subscription WIP: Do not merge - waiting new Inspec release & train update

### PR DESCRIPTION
**Pull request by samscottchef**
Signed-off-by: Sam Scott <sscott@chef.io>

### Description

This resource was requested from here https://github.com/inspec/inspec-aws/issues/145

Also pending discussion around integration tests as this resource is for the advanced shield subscription.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [] New Terraform resources
- [x] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/aws_shield_subscription.md
Comment (omerdemirok) [inline docs/resources/aws_shield_subscription.md:32]: It could be nice to mention that this is an array of items in this format: https://docs.aws.amazon.com/waf/latest/DDOSAPIReference/API_Limit.html

File: docs/resources/aws_shield_subscription.md
Comment (omerdemirok) [inline docs/resources/aws_shield_subscription.md:55]: The text and the URL should be updated.
https://docs.aws.amazon.com/IAM/latest/UserGuide/list_awsshield.html

**Review by omerdemirok — COMMENTED** [review]
Would it be better to rename this resource to `aws_shield_advanced_subscription`?

The current name might mislead users to try testing the standard shield subscription.

CC: @skpaterson, @rmoles.

File: docs/resources/aws_shield_subscription.md
Comment (omerdemirok) [inline docs/resources/aws_shield_subscription.md:8]: I think we should use `shield advanced subscription` instead of `shield subscription` in the whole document.

**Comment by rmoles** [comment]
> Would it be better to rename this resource to `aws_shield_advanced_subscription`?
> 
> The current name might mislead users to try testing the standard shield subscription.
> 
> CC: @skpaterson, @rmoles.

IMO `aws_shield_subscription` is fine. Despite the AWS standard shield being implemented automatically. The fact this resource title contains `subscription` implies that we are testing the paid resource, which would be the advanced shield.

File: docs/resources/aws_shield_subscription.md
Comment (samscottchef) [reply docs/resources/aws_shield_subscription.md]: updated, thanks!

File: docs/resources/aws_shield_subscription.md
Comment (samscottchef) [reply docs/resources/aws_shield_subscription.md]: updated with some more clarification on limits format

**Review by rmoles — APPROVED** [review]
This looks good to me. Holding off merging until a new release of Inspec & train-aws

**Review by omerdemirok — APPROVED** [review]
Thanks, @samscottchef! 
Great work.

**Comment by tbugfinder** [comment]
Hm, what's holding off merging this PR?

**Comment by tbugfinder** [comment]
@omerdemirok May I ask again about the status of this PR?

**Comment by omerdemirok** [comment]
> @omerdemirok May I ask again about the status of this PR?

Hey @tbugfinder, I am not a maintainer of this repository anymore. So, I don't know the answer unfortunately.

**Review by sathish-progress — CHANGES_REQUESTED** [review]
Could you please update Terraform files & complete integration tests for this?

**Comment by sathish-progress** [comment]
This PR requires the `buildkite` to be successfully passed and integration tests are successful before merging..

**Comment by sathish-progress** [comment]
Closing this PR as this is stale for a very long time!  
@samscottchef/ @tbugfinder feel free to re-open this when you are ready with the changes!

**Comment by tbugfinder** [comment]
Was the PR branch deleted by intention?

**Comment by sathish-progress** [comment]
Yes, if you like to contribute to the same branch, I can restore it ..

**Pull request closed without merging**
